backend.native/tests/build.gradle

```diff
@@ -3880,6 +3884,37 @@ if (PlatformInfo.isAppleTarget(project)) {
         interop = 'objcKt34467'
         goldValue = "OK\n42\n"
     }
+
+    interopTest("interop_objc_illegal_sharing_with_weak") {
+        source = "interop/objc/illegal_sharing_with_weak/main.kt"
+        interop = 'objc_illegal_sharing_with_weak'
+
+        expectedExitStatusChecker = { it != 0 }
+        outputChecker = {
+            it.startsWith("Before") &&                   // Should crash after "Before"
+                !it.contains("After") &&                 // But before "After"
+                it.contains("isObjectAliveShouldCrash")  // And should contain stack trace.
```

Comment:
Btw, does this work with `-opt`?

Reply:
Marked the function as `noinline`. Now it definitely should.